core/class/new_spec.rb

```diff
@@ -94,13 +94,11 @@ def message2; "hello"; end
     Class.new.superclass.should == Object
   end
 
-  it "raises a TypeError when given a non-Class" do
-    error_msg = /superclass must be a Class/
-    -> { Class.new("")         }.should raise_error(TypeError, error_msg)
-    -> { Class.new(1)          }.should raise_error(TypeError, error_msg)
-    -> { Class.new(:symbol)    }.should raise_error(TypeError, error_msg)
-    -> { Class.new(mock('o'))  }.should raise_error(TypeError, error_msg)
-    -> { Class.new(Module.new) }.should raise_error(TypeError, error_msg)
+  test_cases = ["", 1, :symbol, mock('o'), Module.new, BasicObject.new].each do |superclass|
+    it "raises a TypeError when given an instance of #{Kernel.instance_method(:class).bind(superclass).call.name}" do
+      error_msg = /superclass must be a Class/
+      -> { Class.new(superclass) }.should raise_error(TypeError, error_msg)
+    end
   end
```

Comment:
Minor issues:
- `test_cases` variable is never used
- `error_msg` variable seems excessive - I would suggest `raise_error(TypeError, /superclass must be a Class/)`
- why do we need `Kernel.instance_method(:class).bind(superclass).call.name` instead of just `superclass.class.name`?
- I noticed the test case description is changed from `it "raises a TypeError when given a non-Class"` to `it "raises a TypeError when given an instance of <class>`. I would keep original description may be as an outer context which wraps all the generated test cases

Reply:
> why do we need Kernel.instance_method(:class).bind(superclass).call.name instead of just superclass.class.name?
Because `BasicObject` doesn't have `class` property, so this way we can access it.